x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/analysis/AnalyzerTests.java

```diff
@@ -267,10 +267,46 @@ public void testNoProjection() {
     }
 
     public void testProjectOrder() {
+        assertProjection("""
```

Comment:
Try to separate all these tests in separate methods. Not necessarily one test per method, but definitely not 10 separate queries in the same test method.
It helps with testing: if the first query fails the rest of them are not even checked.